src/store/actions.ts

```diff
@@ -312,37 +312,47 @@ export const fetchCorpNum = async ({ getters }, corpNum: string): Promise<any> =
 }
 
 // FUTURE: not an action - move it to another module?
-export const checkCOLIN = ({ getters }, corpNum: string) => {
-  const myToken = sessionStorage.getItem('KEYCLOAK_TOKEN')
-
-  // check entity for the corp num first
-  let url = `${appBaseURL}/businesses/${corpNum}`
-  return axios.get(url, {}).catch(error => {
-    if (error.response && error.response.status === 404) {
-      // Remove BC prefix as Colin only supports base number with no prefix for BC's
-      const cleanedCorpNum = corpNum.replace(/^BC+/i, '')
-      url = `${appBaseURL}/colin/${cleanedCorpNum}`
-      return axios.post(url, {}).then(response => {
-        if (response.data.directors === 'Not Available' && response.data.incorporated === 'Not Available') {
-          const error = new Error('Not Found')
-          return Promise.reject(error)
+async function checkCOLIN ({ getters }, corpNum: string): Promise<BusinessFetchIF> {
+  try {
+    // first try to find business in Entities (Legal API)
+    const data = await NamexServices.searchEntities(corpNum)
+    return {
+      identifier: data.identifier,
+      legalName: data.legalName,
+      legalType: data.legalType,
+      state: data.state
+    }
```

Comment:
I previously wrote this in a different syntax but I prefer the "simple" syntax above.

Here's the alternate syntax. Do you understand it?

```
const { identifier, legalName, legalType, state } = await NamexServices.searchEntities(corpNum)
return { identifier, legalName, legalType, state }
```

Reply:
No responses on this? 😃

Reply:
Yes! Much easier to read now 🙌